### ISSUE TITLE
zero value for zerolog.Logger

### ISSUE THREAD

**Issue by soulne4ny** [rs/zerolog#14]
```golang
package main

import (
	"github.com/rs/zerolog"
)

func main() {
	var log zerolog.Logger
	log.Info().Msg("crash")
}
```

```
$ go run zerolog-zero.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x109832b]

goroutine 1 [running]:
github.com/rs/zerolog.(*Event).write(0xc420012240, 0x1b, 0x1f4)
	/.../go/src/github.com/rs/zerolog/event.go:61 +0x6b
github.com/rs/zerolog.(*Event).Msg(0xc420012240, 0x10ca3d8, 0x5)
	/.../go/src/github.com/rs/zerolog/event.go:86 +0x59
main.main()
	/.../go/src/exp/zerolog-zero.go:9 +0x9a
exit status 2
```

The best zero value would be `github.com/rs/zerolog/log` Logger.
Being calm is also better than panic, imho. Or at least it could panic at `Info()` event creation, not at writing `Msg()`.

**Comment by Oxygend** [comment]
It does not seem like you initialized the logger.
Try something like this
```Go
package main

import (
	"github.com/rs/zerolog"
)

func main() {
	log := zerolog.New(os.Stdout).With().Timestamp().Logger()
	log.Info().Msg("crash")
}
```

**Comment by toravir** [comment]
@Oxygend Do you think it is ok to change methods like Info(), Debug() etc to take pointer receivers ? So that one can check for nil pointers and not crash. That will also avoid compiler making a copy of the logger for the invocation.

**Comment by Oxygend** [comment]
@toravir It's zero log, I guess if pointers were used instead, they would cause allocations which we are trying to avoid.

**Comment by toravir** [comment]
@Oxygend i tried that and ran benchmark and i don't see an extra alloc - cpu usage isn't much different either. See:
With "(l * Logger) Info() *Event" : 
```
BenchmarkInfo-8   	30000000	        39.2 ns/op	       0 B/op	       0 allocs/op
```
Without (base code):
```
BenchmarkInfo-8   	30000000	        41.8 ns/op	       0 B/op	       0 allocs/op
```

**Comment by rs** [comment]
Fixed by #18.
